Sources/Workspace/Workspace.swift

```diff
@@ -569,16 +569,18 @@ public class Workspace {
         reset: Bool = false
     ) throws {
         if reset {
-            try pinsStore.unpinAll()
+            pinsStore.unpinAll()
+            try pinsStore.saveState()
         }
         // Start pinning each dependency.
         for dependencyManifest in dependencyManifests.dependencies {
-            try pin(
-                pinsStore: pinsStore,
+            pin(pinsStore: pinsStore,
                 dependency: dependencyManifest.dependency,
                 package: dependencyManifest.manifest.name,
                 reason: reason)
         }
+        
+        try pinsStore.saveState()
     }
 
     /// Pins the managed dependency.
```

Comment:
nit: We should probably mention here that this does *not* saves the pins to disk.